ksql-engine/src/main/java/io/confluent/ksql/function/udtf/array/ExplodeFunctionFactory.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function.udtf.array;
+
+import io.confluent.ksql.function.KsqlTableFunction;
+import io.confluent.ksql.function.TableFunctionFactory;
+import io.confluent.ksql.function.udf.UdfMetadata;
+import io.confluent.ksql.name.FunctionName;
+import io.confluent.ksql.util.KsqlException;
+import java.util.List;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Schema.Type;
+
+public class ExplodeFunctionFactory extends TableFunctionFactory {
+
+  private static final FunctionName NAME = FunctionName.of("EXPLODE");
+
+
+  public ExplodeFunctionFactory() {
+    super(new UdfMetadata(NAME.name(),
+        "",
+        "Confluent",
+        "",
+        "",
+        false));
+  }
+
+  @SuppressWarnings("unchecked")
+  @Override
+  public KsqlTableFunction<?, ?> createTableFunction(final List<Schema> argTypeList) {
+    if (argTypeList.size() != 1) {
+      throw new KsqlException("EXPLODE function should have one arguments.");
+    }
+
+    final Schema schema = argTypeList.get(0);
+    if (schema.type() == Type.ARRAY) {
```

Comment:
This will currently throw when building the logical schema (and it's the same for aggregate functions). It's still before invocation though. This seems like a reasonable place for it. It seems to me the job of the analyzer is to extract basic information from the query, not to do too much.

Reply:
Not sure I agree - we should touch base on this.

Reply:
https://github.com/confluentinc/ksql/issues/3671